packages/core/src/babelPlugins/babel-plugin-redwood-routes-auto-loader.ts

```diff
@@ -25,9 +27,24 @@ export default function (
         if (pages.length === 0) {
           return
         }
+
+        // Remove Page imports in prerender mode (see babel-preset)
+        // This is to make sure that the router receives Pages consistently
+        if (useStaticImports) {
+          // Match import paths, const name could be different
+          const userImportPath = p.node.source?.value
+
+          if (rwPageImportPaths.includes(userImportPath)) {
+            p.remove()
+          }
```

Comment:
Thinking this through. This code is intended to removes manually imported "Page modules"?

So, say a user imports something:

```js
import MyPage from 'src/pages/MyPage'
```

I don't think the path you're receiving in `userImportPath` is the resolved absolute path? It would still be `src/pages/MyPage` But the path in `rwPageImports` paths is an absolute path, so the comparison probably doesn't yield the correct results.

Reply:
I'm not 100% sure, but my gut-understand of this makes me believe this is the case. Maybe a test would be helpful.

My "gut-gut" feeling is that this code isn't really required, unless this is used to make sure you don't double-import something which would cause a syntax-error?

Long term a better strategy might be:

1. Gather all the modules associated to `<Routes page={} />`.
2. Remove them if they're imported.
3. Re-import them.

Reply:
> Long term a better strategy might be:

    Gather all the modules associated to <Routes page={} />.
    Remove them if they're imported.
    Re-import them.

This is what the code here is doing:

1. Check if import statement is a page module. If it is, then remove the import statement.
2. Later in the code, import pages with require

Reply:
I've added a test that illustrates what I was trying to say above.

Reply:
Thanks for showing me this @peterp! I finally understand what you mean, and you're right, the paths don't match when this plugin runs in isolation.

However, the reason I think this works is because `babel-plugin-module-resolver` runs before the routes autoloader plugin. Which means we also get the absolute path in `userImportPath`

Here's an example console logging from this plugin:
```
User import path: /Users/dac09/Code/rw-demo-site/web/src/pages/LandingPage/LandingPage
Redwood Paths: [
  '/Users/dac09/Code/rw-demo-site/web/src/pages/AboutPage/AboutPage',
  '/Users/dac09/Code/rw-demo-site/web/src/pages/DashHomePage/DashHomePage',
  '/Users/dac09/Code/rw-demo-site/web/src/pages/EditTaskPage/EditTaskPage',
  '/Users/dac09/Code/rw-demo-site/web/src/pages/FatalErrorPage/FatalErrorPage',
  '/Users/dac09/Code/rw-demo-site/web/src/pages/LandingPage/LandingPage',
  '/Users/dac09/Code/rw-demo-site/web/src/pages/NewTaskPage/NewTaskPage',
  '/Users/dac09/Code/rw-demo-site/web/src/pages/NotFoundPage/NotFoundPage',
  '/Users/dac09/Code/rw-demo-site/web/src/pages/TaskPage/TaskPage',
  '/Users/dac09/Code/rw-demo-site/web/src/pages/TasksPage/TasksPage'
]
```

Reply:
I'm leaving your test as is, let me know if you're ok with this and I'll revert the change to the test.

Ofcourse because its absolute paths, it'll always fail without the @rw/internal mocks (except on one machine :D)

Reply:
Ah ok, that's comforting. I really don't like having interdependent babel-plugins, but I guess we can't get beyond that at this point? Is there a way that we could independently resolve this?

Totes ok to revert my tests. I was thinking of adding another hook to make this easier to work via tests.

Reply:
I really  appreciate you putting so much effort into this review! 🍔

I'm not sure if its a viable option, but ideally we'd be able to do something like `resolvePath(userImportPath)` that would hook into the resolver plugin(s). 

I'm closing out for the day anyway, so will do the revert tomorrow if you don't get to it before I do :)

Reply:
Updated test with mocks for internal, and added comment about babel-plugin-module-resolver.

I know this isn't ideal (plugins should really work independently, like you said), but we can tackle this as an issue separately if you're keen!

Reply:
Nice! Thanks :)